### PR TITLE
[Feature] Metrics that support memory and object statistics

### PR DESCRIPTION
**Pull request by Astralidea**
## Why I'm doing:
We already have a basic memory and object statistics, now we can also expose the metric for easy use when needed. to query historical data.

## What I'm doing:
Dynamically generate metrics about memory and number of objects.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/memory/MemoryStat.java [hunk 1]
```diff
@@ -22,6 +24,8 @@ public class MemoryStat {
 
     private String counterInfo;
 
+    private Map<String, Long> counterMap;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/memory/MemoryStat.java:27]: plz add comments for this member

File: fe/fe-core/src/main/java/com/starrocks/metric/MetricRepo.java [hunk 2]
```diff
@@ -801,6 +807,44 @@ public Long getValue() {
...
 
+    public static void updateMemoryUsageMetrics() {
+        if (GAUGE_MEMORY_USAGE_STATS.size() ==
+                MemoryUsageTracker.MEMORY_USAGE.values().stream().mapToInt(Map::size).sum()) {
+            return;
+        }
+
+        List<GaugeMetricImpl<Long>> memoryUsageGauges = new ArrayList<>();
+        List<GaugeMetricImpl<Long>> objectCountGauges = new ArrayList<>();
+        MemoryUsageTracker.MEMORY_USAGE.forEach((moduleName, module) -> {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/metric/MetricRepo.java:818]: Should refactor the code `initMemoryMetrics()` to use new memory tracking framework.

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/metric/MetricRepo.java:818]: Sure

**Review by nshangyiming — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**